### DISCUSSION TITLE
Guided Conversation for Chatbot?

### DISCUSSION THREAD

**Discussion by Berna-2000** [FlowiseAI/Flowise#1911] [Ideas]
Hi. I'm actually wondering if we could have the option to have a guided conversation with the chatbot? I am not too sure if I could have done this to begin with, but the idea is that, instead of users typing the questions, they just click some options for the questions. The bot will respond to the question chosen, and each option chosen leads to another set of questions. Anybody know how and if this is possible? Thanks!

**Comment by Jaredude** [comment]
What you're describing is a b-tree user experience which would not need Flowise or even an LLM to make happen since it's just following predefined "if/then" branching.

If you had a hard requirement to use Flowise, you could accomplish a b-tree bot fairly easily by making a RAG bot that interacts with a custom chat interface that would display an initial set of opening questions. When the Flowise flow received the question, you would then query a Vector DB based upon that question and your Vector DB would have a set of questions associated with the supplied question. Then on the custom bot side, you'd need to parse the results and display them as questions the user can click/press to repeat the process. The default chat embed with Flowise will not provide you with the btree interactions you're describing, so you would need to customize the FlowiseChatEmbed code or make your own using another library or making it from scratch.

Hope that helps.
